### ISSUE TITLE
answer_start index in train data

### ISSUE THREAD

**Issue by telukuntla** [naver/biobert-pretrained#3]
In train data, I see that 'answer_start' of the answer(in the context paragraph) is always pointing to the first occurrence of the answer. for example. if the answer word - 'MethPed' is appearing multiple times in the context paragraph, answer_start(which is character index) is pointing to the first occurrence of 'MethPed'. is it more appropriate to point answer_start to that particular instance of  'MethPed' where that particular sentence(which 'MethPed' is present ) is very close to what was asked in the question? please let me know.

**Comment by jhyuklee** [comment]
Hi, @telukuntla

I'll assume you are talking about the BioASQ dataset. The problem occurs as we are trying to train our model based on SQuAD model, which usually has a single gold-standard answer position in training sets. In case of BioASQ, we can also do what you asked for (select sentence first), but just selected the first occurrence to simplify the problem. To select the sentence first and then find an answer out of it, it would require another module besides the BERT, and would also require a label for the correct sentence (which does not exist in BioASQ dataset).

Thank you.
(Next time, we would appreciate if you post fine-tuning issues in https://github.com/dmis-lab/biobert to maintain consistency of this repository.)

**Comment by telukuntla** [comment]
Sorry I should have posted this in https://github.com/dmis-lab/biobert. Thanks a lot for your response.  you are correct, I was talking about BioASQ dataset. and yes we would require additional module to identify sentence and then index of the answer word. please let me know if we take first occurrence of the word(like we are doing it now), will it be any different from identifying sentence first and then index of the answer word, in terms of how BERT(or BioBert) learns Question answering patterns? I mean will BERT(or BioBert) perform even better if we identify sentence first and then index of the answer word or it doesnt matter? please let me know.  (I am sorry next follow up posts will be posting under https://github.com/dmis-lab/biobert.)

**Comment by jhyuklee** [comment]
Yes, if we can correctly identify sentences first, there is a room for improvement. But note that we need a label indicating which sentence corresponds to the best sentence that contains the answer. In case of SQuAD, it has exact answer position in its training/test sets, so it is easy to identify the best sentence. But in case of BioASQ, we do not have such answer position neither in train or test sets. For the further discussion of sentence selection methods, this paper (http://www.aclweb.org/anthology/P18-1160) from ACL might help. Thank you.

**Comment by telukuntla** [comment]
Thank You.
